bcgov/biohubbc
# SIMSBIOHUB-30: SIMS Home Page

# Overview

Creates the publicly-accessible home page in SIMS.

## Links to Jira tickets

 - https://apps.nrs.gov.bc.ca/int/jira/browse/SIMSBIOHUB-30

## Description of relevant changes
 - Moved Resources page from `/admin/resources` to just `/resources`;
 - Added utility functions for building generic URLs and for generating the Login Page URL (historically, we have programatically sent the user to the Keycloak instance);
 - Added a `/login` helper endpoint, which sends the user to the Keycloak instance;
 - When unauthenticated access the SIMS application they land on a home page (historically, they are redirected to the Keycloak instance);
 - Added repo and service classes for Administrative Activities
 - Created the SIMS Landing Page


## Testing Procedure

1. Access the application having not signed in. You should:

- [x] See the SIMS Home page.

2. Click **Resources** in the header. You should:

- [x] Be able to access the Resources page.

3. Click the BioHub logo in the header. You should:

- [x] Arrive back on the Home page.

4. Click the **Log In**. You should:

- [x] Be taken to the Keycloak instance.

5. Proceed to log in to SIMS as an administrator. You should:

- [x] Be redirected to the Projects page.

6. Click the BioHub logo in the header. You should:

 - [x] Have the app root as your browser URL location
 - [x] Arrive back on the Home page;
 - [x] See the message _Welcome back, \<USERNAME\>_;
 - [x] See the **View Projects** button;
 - [x] See the **Manage Users** button;

7. Change your role to a non-administrator, or sign into SIMS as a non-administrator.
8. Navigate back to the home screen. You should:

 - [ ] See the message _Welcome back, \<USERNAME\>_;
 - [ ] See the **View Projects** button;
 - [ ] See the **Create a Project** button;

9. Log out of the application. You should:

 - [ ] Be redirected to the Home page;
